test/ranges.jl

```diff
@@ -1418,7 +1446,6 @@ end
     @test_throws ArgumentError range(nothing)
     @test_throws ArgumentError range(1, step=4)
     @test_throws ArgumentError range(nothing, length=2)
-    @test_throws ArgumentError range(1.0, step=0.25, stop=2.0, length=5)
```

Comment:
The docs say that specifying all four arguments is allowed if they are consistent. So this test contradicts the docs.

Reply:
I decided to keep the behaviour of master here. Since it may be a controversial feature.